pkg/noun/nock.c

```diff
@@ -1924,6 +1927,26 @@ _n_hint_fore(u3_cell hin, u3_noun bus, u3_noun* clu)
       *clu = u3nt(u3k(tag), *clu, now);
     } break;
 
+    case c3__jinx: {
+      if (c3y == u3a_is_atom(*clu)) {
+        // clu is in Urbit time, but we need Unix time
+        mpz_t clu_mp;
+        u3r_mp(clu_mp, *clu);
+        mpz_t urs_mp, tim_mp;
+        mpz_init(urs_mp);
+        mpz_init(tim_mp);
+        mpz_tdiv_q_2exp(tim_mp, clu_mp, 48);
+        mpz_mul_ui(tim_mp, tim_mp, 1000);
+        mpz_tdiv_q_2exp(urs_mp, tim_mp, 16);
+        c3_w mil_w = u3i_mp(urs_mp);
+        u3m_timer_set(mil_w); // set ITIMER (mil_w is in microseconds)
+        mpz_clear(clu_mp);
+        mpz_clear(tim_mp);
+      }
```

Comment:
You can replace this math with `u3_time_msc_out(c3_d)`.

```
c3_d tim_d;
if ( c3y == u3r_safe_chub(*clu, &tim_d) ) {
  c3_w mic_w = u3_time_msc_out(tim_d);
  c3_w mil_w = mic_w / 1000;
  u3m_timer_set(mil_w);
}
```

(Or you could, if it was defined in `pkg/vere`. I'd be inclined to copy it into this file as a static function rather than use gmp.)

Reply:
Urbit time is bigger than that, though, so the `u3r_safe_chub` always fails.

Reply:
It could retrieve two chubs for the time, very unlikely to be too big for that.